// react-canvas-confetti/dist/conductor/vortex/index.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var index_1 = __importDefault(require("../index"));
var angels = [0, -45, -90, -135, -180, -225, -270, -315];
var VortexConductor = (function (_super) {
    __extends(VortexConductor, _super);
    function VortexConductor() {
        var _this = _super.apply(this, __spreadArray([], __read(arguments), false)) || this;
        _this.tickCount = 0;
        _this.tickAnimation = function () {
            _this.confetti(_this.decorateOptions({
                spread: 120,
                ticks: 60,
                gravity: 0,
                decay: 0.94,
                startVelocity: 20,
                colors: ["004e64", "00a5cf", "#9fffcb", "#25a18e", "#7ae582"],
                particleCount: 60,
                shapes: ["circle", "square"],
                angle: _this.angle,
            }));
        };
        return _this;
    }
    Object.defineProperty(VortexConductor.prototype, "angle", {
        get: function () {
            return angels[this.tickCount++ % angels.length];
        },
        enumerable: false,
        configurable: true
    });
    return VortexConductor;
}(index_1.default));
exports.default = VortexConductor;
