// rsuite/esm/InlineEdit/InlineEdit.js
'use client';
import _extends from "@babel/runtime/helpers/esm/extends";
import _objectWithoutPropertiesLoose from "@babel/runtime/helpers/esm/objectWithoutPropertiesLoose";
import React from 'react';
import PropTypes from 'prop-types';
import { useClassNames, mergeRefs } from '../utils';
import { oneOf } from '../internals/propTypes';
import EditableControls from './EditableControls';
import useFocusEvent from './useFocusEvent';
import useEditState from './useEditState';
import { renderChildren, defaultRenderInput } from './renderChildren';
var InlineEdit = /*#__PURE__*/React.forwardRef(function (props, ref) {
  var _props$as = props.as,
    Component = _props$as === void 0 ? 'div' : _props$as,
    _props$children = props.children,
    children = _props$children === void 0 ? defaultRenderInput : _props$children,
    _props$classPrefix = props.classPrefix,
    classPrefix = _props$classPrefix === void 0 ? 'inline-edit' : _props$classPrefix,
    className = props.className,
    disabled = props.disabled,
    size = props.size,
    _props$showControls = props.showControls,
    showControls = _props$showControls === void 0 ? true : _props$showControls,
    _props$stateOnBlur = props.stateOnBlur,
    stateOnBlur = _props$stateOnBlur === void 0 ? 'save' : _props$stateOnBlur,
    placeholder = props.placeholder,
    rest = _objectWithoutPropertiesLoose(props, ["as", "children", "classPrefix", "className", "disabled", "size", "showControls", "stateOnBlur", "placeholder"]);
  var _useClassNames = useClassNames(classPrefix),
    withClassPrefix = _useClassNames.withClassPrefix,
    merge = _useClassNames.merge,
    prefix = _useClassNames.prefix;
  var _useEditState = useEditState(_extends({}, rest, {
      disabled: disabled
    })),
    value = _useEditState.value,
    isEditing = _useEditState.isEditing,
    onSave = _useEditState.onSave,
    onCancel = _useEditState.onCancel,
    onChange = _useEditState.onChange,
    onKeyDown = _useEditState.onKeyDown,
    onClick = _useEditState.onClick,
    htmlProps = _useEditState.htmlProps;
  var _useFocusEvent = useFocusEvent({
      isEditing: isEditing,
      stateOnBlur: stateOnBlur,
      onSave: onSave,
      onCancel: onCancel
    }),
    target = _useFocusEvent.target,
    root = _useFocusEvent.root,
    onBlur = _useFocusEvent.onBlur;
  var childrenProps = {
    size: size,
    value: value,
    disabled: disabled,
    placeholder: placeholder,
    plaintext: !isEditing,
    onChange: onChange,
    onBlur: onBlur
  };
  return /*#__PURE__*/React.createElement(Component, _extends({
    ref: mergeRefs(root, ref),
    tabIndex: 0,
    className: merge(className, withClassPrefix(size, {
      disabled: disabled
    })),
    onClick: onClick,
    onKeyDown: onKeyDown
  }, htmlProps), renderChildren(children, childrenProps, target), showControls && isEditing && /*#__PURE__*/React.createElement(EditableControls, {
    className: prefix('controls'),
    onSave: onSave,
    onCancel: onCancel
  }));
});
InlineEdit.displayName = 'InlineEdit';
InlineEdit.propTypes = {
  children: PropTypes.any,
  classPrefix: PropTypes.string,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  defaultValue: PropTypes.any,
  value: PropTypes.any,
  showControls: PropTypes.bool,
  placeholder: PropTypes.string,
  size: oneOf(['lg', 'md', 'sm', 'xs']),
  stateOnBlur: oneOf(['save', 'cancel']),
  onChange: PropTypes.func,
  onCancel: PropTypes.func,
  onSave: PropTypes.func,
  onEdit: PropTypes.func
};
export default InlineEdit;
