// react-canvas-confetti/dist/conductor/photons/index.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var index_1 = __importDefault(require("../index"));
var randomInRange_1 = __importDefault(require("../../helpers/randomInRange"));
var colors = [
    "#f94144",
    "f3722c",
    "f8961e",
    "f9c74f",
    "90be6d",
    "43aa8b",
    "577590",
];
var config = [
    {
        origin: function () { return ({ x: (0, randomInRange_1.default)(0, 1), y: -0.1 }); },
        angle: function () { return (0, randomInRange_1.default)(0, -180); },
    },
    {
        origin: function () { return ({ x: (0, randomInRange_1.default)(0, 1), y: 1.1 }); },
        angle: function () { return (0, randomInRange_1.default)(0, 180); },
    },
    {
        origin: function () { return ({ x: -0.1, y: (0, randomInRange_1.default)(0, 1) }); },
        angle: function () { return (0, randomInRange_1.default)(-90, 90); },
    },
    {
        origin: function () { return ({ x: 1.1, y: (0, randomInRange_1.default)(0, 1) }); },
        angle: function () { return (0, randomInRange_1.default)(90, 270); },
    },
];
var PhotonsConductor = (function (_super) {
    __extends(PhotonsConductor, _super);
    function PhotonsConductor() {
        var _this = _super.apply(this, __spreadArray([], __read(arguments), false)) || this;
        _this.tickCount = 0;
        _this.tickAnimation = function () {
            var colorIndex = Number((0, randomInRange_1.default)(0, colors.length - 1).toFixed());
            var _a = _this.params, angle = _a.angle, origin = _a.origin;
            _this.confetti(_this.decorateOptions({
                particleCount: 1,
                angle: angle(),
                spread: 0,
                gravity: 0,
                ticks: 600,
                decay: 1,
                startVelocity: 7,
                flat: true,
                origin: origin(),
                shapes: ["circle"],
                scalar: (0, randomInRange_1.default)(0.2, 6),
                colors: [colors[colorIndex]],
            }));
        };
        return _this;
    }
    Object.defineProperty(PhotonsConductor.prototype, "params", {
        get: function () {
            return config[this.tickCount++ % config.length];
        },
        enumerable: false,
        configurable: true
    });
    return PhotonsConductor;
}(index_1.default));
exports.default = PhotonsConductor;
