// react-canvas-confetti/dist/conductor/crossfire/index.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
var __spreadArray = (this && this.__spreadArray) || function (to, from, pack) {
    if (pack || arguments.length === 2) for (var i = 0, l = from.length, ar; i < l; i++) {
        if (ar || !(i in from)) {
            if (!ar) ar = Array.prototype.slice.call(from, 0, i);
            ar[i] = from[i];
        }
    }
    return to.concat(ar || Array.prototype.slice.call(from));
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var index_1 = __importDefault(require("../index"));
var randomInRange_1 = __importDefault(require("../../helpers/randomInRange"));
var CrossfireConductor = (function (_super) {
    __extends(CrossfireConductor, _super);
    function CrossfireConductor() {
        var _this = _super.apply(this, __spreadArray([], __read(arguments), false)) || this;
        _this.tickAnimation = function () {
            var gravity = 0;
            var colors = ["#E8B837"];
            var particleCount = (0, randomInRange_1.default)(13, 17);
            var spread = (0, randomInRange_1.default)(75, 85);
            var decay = (0, randomInRange_1.default)(0.97, 0.99);
            var startVelocity = (0, randomInRange_1.default)(9, 11);
            var ticks = (0, randomInRange_1.default)(40, 60);
            _this.confetti(_this.decorateOptions({
                particleCount: particleCount,
                spread: spread,
                colors: colors,
                decay: decay,
                startVelocity: startVelocity,
                ticks: ticks,
                gravity: gravity,
                angle: 45,
                origin: { x: 0, y: 1 },
            }));
            _this.confetti(_this.decorateOptions({
                particleCount: particleCount,
                spread: spread,
                colors: colors,
                decay: decay,
                startVelocity: startVelocity,
                ticks: ticks,
                gravity: gravity,
                angle: -45,
                origin: { x: 0, y: 0 },
            }));
            _this.confetti(_this.decorateOptions({
                particleCount: particleCount,
                spread: spread,
                colors: colors,
                decay: decay,
                startVelocity: startVelocity,
                ticks: ticks,
                gravity: gravity,
                angle: -135,
                origin: { x: 1, y: 0 },
            }));
            _this.confetti(_this.decorateOptions({
                particleCount: particleCount,
                spread: spread,
                colors: colors,
                decay: decay,
                startVelocity: startVelocity,
                ticks: ticks,
                gravity: gravity,
                angle: 135,
                origin: { x: 1, y: 1 },
            }));
        };
        return _this;
    }
    return CrossfireConductor;
}(index_1.default));
exports.default = CrossfireConductor;
